src/tr2/game/game_flow/sequencer_events.c

```diff
@@ -114,23 +115,29 @@ static DECLARE_GF_EVENT_HANDLER(M_HandlePlayLevel)
 
     switch (seq_ctx) {
     case GFSC_SAVED:
-        ExtractSaveGameInfo();
+        const int16_t slot_num = Savegame_GetBoundSlot();
+        if (!Savegame_Load(slot_num)) {
+            LOG_ERROR("Failed to load save file!");
+            Game_SetCurrentLevel(nullptr);
+            GF_SetCurrentLevel(nullptr);
```

Comment:
Are these two setters necessary?

Reply:
Good point, it seems to be fine without it. I can follow up with TR1 (it's the same there) after this PR while doing some refactoring on the SG logic there to do the same.